Code/Framework/AzCore/AzCore/Instance/InstancePool.h

```diff
@@ -98,9 +106,10 @@ namespace AZ
 
         // creates a pool with the explicit name given, and the reset function specified
         template<class T>
-        AZ::Outcome<AZStd::shared_ptr<InstancePool<T>>, AZStd::string> CreatePool(AZ::Name name, typename InstancePool<T>::ResetInstance resetFunc)
+        AZ::Outcome<AZStd::shared_ptr<InstancePool<T>>, AZStd::string> CreatePool(
+            AZ::Name name, typename InstancePool<T>::ResetInstance resetFunc = [](T&){}, typename InstancePool<T>::CreateInstance createFunc = []() { return new T; })
```

Comment:
Can you make this default to `return new T{}`?

This will perform `value initialization` for types that don't have constructor which will zero initialize it's members instead of leaving it in a indeterminate state using `default initilaization

Reply:
yep, happy to. Done now :)